### PR TITLE
bug / OS-771 : fix bug erc20 type collision

### PR DESCRIPTION
**Pull request by Rekard0**
## Description

QA: at https://subgraph.satsuma-prod.com/aragon/osx-baseGoerli/playground
query:
`
{
 erc20Transfers {
    id
    token {
      id
      __typename
    }
  }
}
`

Task ID: [OS-771](https://aragonassociation.atlassian.net/browse/OS-771)

## Type of change
See the framework lifecylce in `packages/contracts/docs/framework-lifecycle` to decide what kind of change this pull request is.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.

[OS-771]: https://aragonassociation.atlassian.net/browse/OS-771?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/src/utils/tokens/erc20.ts [hunk 1]
```diff
@@ -95,6 +95,31 @@ export function fetchERC20(address: Address): ERC20Contract | null {
   return contract;
 }
 
+/**
+ * @dev Determines the type of ERC20 token (wrapped or regular) and returns its address.
+ *
+ * @param token The address of the token to be determined.
+ * @return tokenAddress The address of the ERC20 token if it's either wrapped or regular, null otherwise.
+ */
+export function determineERC20Token(token: Address): string | null {
```
Comment (Michael-A-Heuer) [inline packages/subgraph/src/utils/tokens/erc20.ts:104]: `determineERC20Token` might not be a telling name. How about `getUnderlyingERC20TokenAddress`?

File: packages/subgraph/src/utils/tokens/erc20.ts [hunk 2]
```diff
@@ -95,6 +95,31 @@ export function fetchERC20(address: Address): ERC20Contract | null {
...
+  let tokenAddress: string;
+  if (supportsERC20Wrapped(token)) {
+    let contract = fetchWrappedERC20(token);
+    if (!contract) {
+      return null;
+    }
+    tokenAddress = contract.id;
```
Comment (Michael-A-Heuer) [inline packages/subgraph/src/utils/tokens/erc20.ts:111]: Unrelated to this PR, but should token have maybe a separate `address` field that is then also returned as the ID? Seems cleaner to me.
Ignore this comment if this is done on purpose for performance reasons.

**Review by Michael-A-Heuer — APPROVED** [review]
Small naming suggestion. Otherwise, LGTM

File: packages/subgraph/src/utils/tokens/erc20.ts (hunk 1 above)
Comment (josemarinas) [reply packages/subgraph/src/utils/tokens/erc20.ts:104]: @Michael-A-Heuer  I dont think thats a good name either, the function checks if a token is wrapped or not and returns the instance id

File: packages/subgraph/schema.graphql [hunk 3]
```diff
@@ -104,7 +104,7 @@ interface TokenTransfer {
 type ERC20Transfer implements TokenTransfer @entity(immutable: true) {
   id: ID!
   dao: Dao!
-  token: ERC20Contract!
+  token: Token!
```
Comment (josemarinas) [inline packages/subgraph/schema.graphql:107]: This shouldn't affect ths SDK as long as as it contains this

```
id
name
symbol
decimals
```
I believe it does so no problem with that

**Review by josemarinas — DISMISSED** [review]
If you can confirm that token has the specified fields, its fine for me

File: packages/subgraph/src/utils/tokens/erc20.ts (hunk 1 above)
Comment (Rekard0) [reply packages/subgraph/src/utils/tokens/erc20.ts]: addressed in https://github.com/aragon/osx/pull/484/commits/945c0fb413f87f0ec138e5e03e7339367e1f9901

**Review by josemarinas — APPROVED** [review]
Looks good, I would like a shorter name for the functions but it adds clarity, If @Michael-A-Heuer  likes it I'm good

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**